### PR TITLE
setup directory v3 client

### PR DESCRIPTION
**Pull request by gimmyxd**
Add Directory v3

- run unit tests: `bx rake spec:unit`
- run integration tests(need to have topaz installed): `bx rake spec:integration`
- to see the generated docs run `yard doc && yard server`

### CODE REVIEW

File: spec/integration/topaz.rb [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    # 2 minutes
+    WAIT_FOR_TOPAZ = 2 * 60
+    CERT_FILE = File.join(ENV.fetch("HOME", ""), ".config/topaz/certs/grpc-ca.crt")
+    DB_DIR = File.join(ENV.fetch("HOME", ""), ".config/topaz/db")
+
+    def run
+      stop
+
+      if File.exist?(File.join(DB_DIR, "directory.db"))
+        File.rename(File.join(DB_DIR, "directory.db"), File.join(DB_DIR, "directory.bak"))
```
Comment (BogdanIrimie) [inline spec/integration/topaz.rb:16]: Should we make a backup for the config as well `cfg/config.yaml`? When we run the `configure` the old config will get lost I think

Comment (gimmyxd) [reply spec/integration/topaz.rb:16]: Yes, I think that is a good idea.

Comment (gimmyxd) [reply spec/integration/topaz.rb:16]: Updated

File: lib/aserto/directory/v3/exporter.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+module Aserto
+  module Directory
+    module V3
+      module Exporter
+        DATA_TYPE = {
+          unknown: 0x0,
+          objects: 0x8,
+          relations: 0x10,
+          all: 0x18
+        }.freeze
```
Comment (carabasdaniel) [inline lib/aserto/directory/v3/exporter.rb:12]: Can't you use https://github.com/aserto-dev/ruby-directory/blob/main/generated/aserto/directory/exporter/v3/exporter_pb.rb#L45 ?

File: lib/aserto/directory/v3/reader.rb [hunk 3]
```diff
@@ -0,0 +1,196 @@
+# frozen_string_literal: true
+
+module Aserto
+  module Directory
+    module V3
+      module Reader
```
Comment (carabasdaniel) [inline lib/aserto/directory/v3/reader.rb:6]: The reader seems to be missing the simple Check call

File: lib/aserto/directory/v3/reader.rb [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+        #
+        # @example
+        #   client.get_relation(
+        #     object_type: "user",
+        #     object_id: "rick@the-citadel.com",
+        #     relation: "member",
+        #     object_type: "group",
+        #     object_id: "evil_genius"
+        #   )
+        def get_relation(object_type:, object_id:, relation:, subject_type:, subject_id:, subject_relation: "",
```
Comment (oanatmaria) [inline lib/aserto/directory/v3/reader.rb:74]: should we also add `get_graph` method?

Comment (gimmyxd) [reply lib/aserto/directory/v3/reader.rb:74]: Done

**Pull request merged**